src/pages/studyView/UserSelections.tsx

```diff
@@ -600,6 +608,66 @@ export default class UserSelections extends React.Component<
         );
     }
 
+    private renderGenePanelFilter(filters: GenePanelFilter[]): JSX.Element {
+        const groupedPanels = filters
+            .map(f => {
+                const profile = f.molecularProfileIds[0].split('_');
+                return {
+                    suffix: profile[profile.length - 1],
+                    filter: f,
+                };
+            })
+            .reduce((prev, cur) => {
+                if (prev.has(cur.suffix)) {
+                    prev.set(
+                        cur.suffix,
+                        prev.get(cur.suffix)!.concat(cur.filter)
+                    );
+                } else {
+                    prev.set(cur.suffix, [cur.filter]);
+                }
+                return prev;
+            }, new Map<string, GenePanelFilter[]>());
+
+        const pills = Array.from(groupedPanels.entries()).map(entry => {
+            const components = entry[1].map(gFilter => {
+                return (
+                    <PillTag
+                        content={gFilter.genePanel}
+                        onDelete={() =>
+                            this.props.removeGenePanelFilter(gFilter)
+                        }
+                        backgroundColor={
+                            STUDY_VIEW_CONFIG.colors.theme.clinicalFilterContent
+                        }
+                    />
+                );
+            });
+            const label = (
+                <span className={styles.filterClinicalAttrName}>
+                    {'Gene Panels: ' + entry[0] + ''}
+                </span>
+            );
+            components.unshift(label);
```

Comment:
don't love unshift here.  maybe start with the label and push to the array in forEach above